docs/api.md

```diff
@@ -85,3 +88,147 @@ components are exported as the public interface for Garden.
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { HTMLAttributes, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import { StyledSubElement } from '../styled';
+
+/* [1] */ export interface IElementSubElementProps extends HTMLAttributes<HTMLElement> {
+  /** Applies regular (non-bold) font weight */
+  isRegular?: boolean;
+}
+
+/* [2] */ const SubElementComponent = forwardRef<IElementSubElementProps>((props, ref) => (
+  <StyledSubElement ref={ref} {...props} />
+));
+
+/* [3] */ SubElementComponent.displayName = 'Element.SubElement';
+
+/* [4] */ SubElementComponent.propTypes = {
+  isRegular: PropTypes.bool
+};
+
+/**
+ * @extends HTMLAttributes<HTMLElement>
+ */
+/* [5] */ export const SubElement = SubElementComponent;
+```
+
+1. The exported interface combines the name of the complex parent element
+   together with the `SubElement`
+2. The functional component uses _XxxComponent_ naming. The `const` is not
+   exported due to a `displayName` [issue with
+   Storybook](https://github.com/storybookjs/storybook/issues/12263#issuecomment-1008870685).
+3. Use the `Element.SubElement` string construct for consistent subcomponent
+   `displayName` naming
+4. All API props should be set as `PropTypes` for runtime error checking
+5. The `@extends` clause is always applied to the export. This export
+   hack-around allows subcomponent props to display in Storybook.
+
+#### Complex
+
+The following typed structure represents the layout for complex Garden
+components – those with subcomponents as static properties. The annotations
+listed below document areas of interest.
+
+```tsx
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { HTMLAttributes, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+/* [1] */ import { SubElement } from './SubElement';
+import { StyledElement } from '../styled';
+
+export interface IElementProps extends HTMLAttributes<HTMLElement> {
+  /** Applies compact styling */
+  isCompact?: boolean;
+}
+
+/* [2] */ const ElementComponent = forwardRef<HTMLElement, IElementProps>((props, ref) => (
+  <StyledElement ref={ref} {...props} />
+));
+
+/* [3] */ ElementComponent.displayName = 'Element';
+
+/* [4] */ ElementComponent.propTypes = {
+  isCompact: PropTypes.bool
+};
+
+/**
+ * @extends HTMLAttributes<HTMLElement>
+ */
+/* [5] */ export const Element = ElementComponent as typeof ElementComponent & {
+  SubElement: typeof SubElement;
+};
+
+/* [6] */ Element.SubElement = SubElement;
+```
+
+1. Subcomponents should each reside in separate files. In Garden, one component
+   = one file
+2. The non-exported `const` is a React functional component and uses _XxxComponent_
+   naming style
+3. The `displayName` definition should be set to the exported component name and
+   satisfies linter checks for runtime naming
```

Comment:
The only exported component is `Element` – so should be the `displayName`. I'll take another crack at the verbiage here.

Reply:
The only exported component is `Element` – so that should be the `displayName`. I'll take another crack at the verbiage here.

Reply:
Hm, I may have taken it too literally with how it is worded? The linting work correctly so it shouldn't be an issue... I just understood it as (example):

```
/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Accordion = AccordionComponent as typeof AccordionComponent & {
  ...
};

Accordion.displayName = 'Accordion';
```

Instead of what we are actually doing:

```
AccordionComponent.displayName = 'Accordion';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Accordion = AccordionComponent as typeof AccordionComponent & {
  ...
};
```

Since `Accordion is what is being exported. Possibly making a distinction between `element`'s and `styled-components` would help.. not too sure frankly.

Reply:
Your (example) can't happen – it breaks both displayName linting and JSDoc parsing.

We actually need the "actually doing" version. And the point is to highlight the necessary `AccordionComponent.displayName = 'Accordion';` mismatch – which satisfies both linting and provides the necessary connective tissue for JSDoc parsing via `garden cmd-docgen`.